### PR TITLE
net: lib: nrf_cloud: improve function comments and nrf_cloud_uninit

### PR DESCRIPTION
**Pull request by jayteemo**
Improve documentation for nrf_cloud functions.
Update nrf_cloud_uninit to wait for the connection poll thread to become inactive.
IRIS-4917

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]
LGTM

**Review by tony-le-24 — APPROVED** [review]
LGTM!

File: include/net/nrf_cloud.h [hunk 1]
```diff
@@ -535,7 +535,7 @@ struct nrf_cloud_init_param {
  *
  * @retval 0       If successful.
  * @retval -EACCES Already initialized or @ref nrf_cloud_uninit is in progress.
- *                 Otherwise, a (negative) error code is returned.
+ * @return A negative value indicates an error.
```
Comment (glarsennordic) [inline include/net/nrf_cloud.h:538]: Is it feasible to list the possible error values here?

File: include/net/nrf_cloud.h [hunk 2]
```diff
@@ -645,13 +650,13 @@ int nrf_cloud_send(const struct nrf_cloud_tx_data *msg);
...
- * If the API succeeds, you can expect the
- * @ref NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED event.
+ * The @ref NRF_CLOUD_EVT_TRANSPORT_DISCONNECTED event indicates
+ * that the disconnect has completed successfully.
  *
  * @retval 0       If successful.
  * @retval -EACCES Cloud connection is not established; wait for
  *                 @ref NRF_CLOUD_EVT_TRANSPORT_CONNECTED.
- *                 Otherwise, a (negative) error code is returned.
+ * @return A negative value indicates an error.
```
Comment (glarsennordic) [inline include/net/nrf_cloud.h:659]: What is the effect of having both return and retval? You do this in quite a few, so I am assuming it is intentional, but I don't understand it

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c [hunk 3]
```diff
@@ -168,6 +168,24 @@ int nrf_cloud_uninit(void)
 		}
 	}
 
+#if IS_ENABLED(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD)
```
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud.c:171]: This seems like it should be its own commit

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c [hunk 4]
```diff
@@ -168,6 +168,24 @@ int nrf_cloud_uninit(void)
...
+	if (atomic_get(&connection_poll_active)) {
+		uint32_t wait_cnt = 50;
+
+		LOG_DBG("Waiting for connection poll thread to become inactive");
+		while (--wait_cnt && atomic_get(&connection_poll_active)) {
```
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud.c:176]: Would a semaphore with a timeout make more sense than waiting on an atomic like this?

**Review by glarsennordic — APPROVED** [review]
LGTM, consider my suggestions optional

File: include/net/nrf_cloud.h (hunk 2 above)
Comment (jayteemo) [reply include/net/nrf_cloud.h:659]: you can see examples of both:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/networking/nrf_cloud.html

this ways looks better imo, but i guess @peknis has the final say.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud.c (hunk 4 above)
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud.c:176]: i don't really like polling on the value, but it didn't seem necessary to add a new semaphore 🤷

File: include/net/nrf_cloud.h (hunk 1 above)
Comment (jayteemo) [reply include/net/nrf_cloud.h:538]: i don't think so, but you can look at the code... maybe you would disagree?

Comment (glarsennordic) [reply include/net/nrf_cloud.h:538]: I'm willing to take your word for it if you say it's not feasible

**Review by peknis — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
CI passed on rerun.
